src/PattyServer.js

```diff
@@ -76,18 +76,28 @@ class PattyServer {
    * @private
    */
   _setProcessOwner() {
-    // nothing to do for non-posix system
-    if (!process.setuid) { return; }
-
     // no target user to switch to
     if (!this.options.processOwner) { return; }
 
+    // set user-id
+    // nothing to do for non-posix system
+    if (!process.setuid) { return; }
```

Comment:
out of curiosity as both return, a reason in the change of order ?

Reply:
just to group the logic:
- 1x check if there ids anything to do
- 2x check if the function exists, then call it